### PR TITLE
Fixes TextArea warnings

### PR DESCRIPTION
**Pull request by cugone**
Fixes int to uint8_t conversion, possible loss of data

### CODE REVIEW

File: src/UI/Core/TextArea.h [hunk 1]
```diff
@@ -14,6 +14,7 @@ class TextArea : public Control
 	TextArea() = default;
 	virtual ~TextArea() = default;
 
+	void textColor(int r, int g, int b, int a = 255) { textColor(NAS2D::Color{static_cast<uint8_t>(r), static_cast<uint8_t>(g), static_cast<uint8_t>(b), static_cast<uint8_t>(a)}); };
```
Comment (DanRStevens) [inline src/UI/Core/TextArea.h]: Is there a way to remove the use of `static_cast`?

Comment (cugone) [reply src/UI/Core/TextArea.h]: No. char/unsigned char/signed char all require down casts.

I wanted to remove it entirely, but that broke a call in `FactoryProduction.cpp`. Decided to leave it as a separate issue.

Comment (DanRStevens) [reply src/UI/Core/TextArea.h]: Yes, I was thinking update the call. Is it just one place?

Comment (cugone) [reply src/UI/Core/TextArea.h]: As far as the compiler warned me it is.

**Pull request merged**